snap/snapcraft.yaml

```diff
@@ -207,6 +222,45 @@ parts:
       - libseccomp-dev
       - make
 
+  nvidia-container-toolkit:
+    plugin: make
+    source: https://github.com/NVIDIA/nvidia-container-toolkit.git
+    source-tag: v1.12.0
+    source-depth: 1
+    override-build: |
+      PLATFORM_DIST="ubuntu18.04"
+
+      make ${PLATFORM_DIST}
+
+      dpkg-deb -x ${SNAPCRAFT_PART_BUILD}/dist/${PLATFORM_DIST}/${SNAPCRAFT_TARGET_ARCH}/nvidia-container-toolkit_*.deb ${SNAPCRAFT_PART_INSTALL}
+      dpkg-deb -x ${SNAPCRAFT_PART_BUILD}/dist/${PLATFORM_DIST}/${SNAPCRAFT_TARGET_ARCH}/nvidia-container-toolkit-base_*.deb ${SNAPCRAFT_PART_INSTALL}
+
+      # From nvidia-container-toolkit package postinst
+      (cd "$SNAPCRAFT_PART_INSTALL/usr/bin" && ln -s nvidia-container-runtime-hook nvidia-container-toolkit)
+    build-packages:
+      - make
+    # Recursive - could also use docker.io build-packages #
+    build-snaps:
+      - docker
+
+  libnvidia-container:
+    plugin: make
+    source: https://github.com/NVIDIA/libnvidia-container.git
+    source-tag: v1.12.0
+    source-depth: 1
+    override-build: |
+      PLATFORM_DIST="ubuntu18.04"
+
+      make ${PLATFORM_DIST}
+
+      dpkg-deb -x ${SNAPCRAFT_PART_BUILD}/dist/${PLATFORM_DIST}/${SNAPCRAFT_TARGET_ARCH}/libnvidia-container1_*.deb ${SNAPCRAFT_PART_INSTALL}
+      dpkg-deb -x ${SNAPCRAFT_PART_BUILD}/dist/${PLATFORM_DIST}/${SNAPCRAFT_TARGET_ARCH}/libnvidia-container-tools_*.deb ${SNAPCRAFT_PART_INSTALL}
+    build-packages:
+      - make
+    # Recursive - could also use docker.io build-packages #
+    build-snaps:
+      - docker
```

Comment:
Also interesting that lxd does not use nvidia-ctk bur only uses nvidia-container-cli

Reply:
Sure - I will take a look at that, and see what's possible :+1:

Reply:
> Also interesting that lxd does not use nvidia-ctk bur only uses nvidia-container-cli

So far, everything I've included is pretty much just according to the various documentation I've found on the NIVIDIA and Docker web sites.  It's possible it could be paired down a bit and still work, but following documented installation seems like a good starting point at least.

Reply:
I think this thread was resolved by my changes yesterday.